diff --git a/proto/mpp.proto b/proto/mpp.proto
index f8af03b..1582074 100644
--- a/proto/mpp.proto
+++ b/proto/mpp.proto
@@ -11,12 +11,24 @@ option (gogoproto.unmarshaler_all) = true;
 
 option java_package = "org.tikv.kvproto";
 
+// Data compression method
+enum CompressMethod {
+  NONE = 0;
+  LZ4 = 1;
+  ZSTD = 2;
+}
+
+message  ExchangeSenderMeta {
+	CompressMethod compress = 1;
+}
+
 // TaskMeta contains meta of a mpp plan, including query's ts and task address.
 message TaskMeta {
 	uint64 start_ts = 1; // start ts of a query
 	int64 task_id = 2; // if task id is -1 , it indicates a tidb task.
 	int64 partition_id = 3; // Only used for hash partition
 	string address = 4; // target address of this task.
+	int64 mpp_version = 5; // mpp version
 }
 
 message IsAliveRequest {
@@ -24,6 +36,7 @@ message IsAliveRequest {
 
 message IsAliveResponse {
     bool available = 1;
+    int64 mpp_version = 2;
 }
 
 // Dipsatch the task request to different tiflash servers.
@@ -36,6 +49,7 @@ message DispatchTaskRequest {
 	int64 schema_ver = 5;
 	// Used for partition table scan
 	repeated coprocessor.TableRegions table_regions = 6;
+    optional ExchangeSenderMeta exchange_sender_meta = 7;
 }
 
 // Get response of DispatchTaskRequest.
@@ -67,6 +81,7 @@ message MPPDataPacket {
 	Error error = 2;
 	repeated bytes chunks = 3;
 	repeated uint64 stream_ids = 4;
+	CompressMethod compress = 5;
 }
 
 message Error {
